presto-main/src/main/java/com/facebook/presto/metadata/FunctionManager.java

```diff
@@ -271,6 +273,10 @@ public static QualifiedFunctionName qualifyFunctionName(QualifiedName name)
      */
     public FunctionHandle resolveFunction(Optional<Map<String, String>> sessionFunctions, Optional<TransactionId> transactionId, QualifiedFunctionName functionName, List<TypeSignatureProvider> parameterTypes)
     {
+        Optional<FunctionHandle> sessionFunctionHandle = sessionFunctionManager.getFunctionHandle(functionName, sessionFunctions);
```

Comment:
You are by passing function resolution here, which is probably why you need to handle coercion separately. All function resolution should go through the same logic, which is wrapped in `resolveFunctionInternal`. Function namespaces are only providing candidates with `getFunction(transactionID, functionName)`.

Reply:
Outdated